src/mds/CInode.h

```diff
@@ -76,12 +77,13 @@ class InodeStore {
   std::string                symlink;      // symlink dest, if symlink
   std::map<std::string, bufferptr> xattrs;
   fragtree_t                 dirfragtree;  // dir frag tree, if any.  always consistent with our dirfrag map.
-  std::map<snapid_t, old_inode_t> old_inodes;   // key = last, value.first = first
-  bufferlist		     snap_blob;    // Encoded copy of SnapRealm, because we can't
+  compact_map<snapid_t, old_inode_t> old_inodes;   // key = last, value.first = first
+  bufferlist		     *_snap_blob;   // Encoded copy of SnapRealm, because we can't
```

Comment:
It feels odd to have a leading underscore on a public member like this.  Actually, it seems like the only out-of-class access to this is from CDir::_omap_fetched, so what about making this private and just having a default decode_snap_blob() method that reads out the private _snap_blob.